plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/statistics/DeltaLakeJsonFileStatistics.java

```diff
@@ -175,7 +177,18 @@ private static Long readStatisticsTimestamp(String timestamp)
 
     private static Long readStatisticsTimestampWithZone(String timestamp)
     {
-        ZonedDateTime zonedDateTime = ZonedDateTime.parse(timestamp, JSON_STATISTICS_TIMESTAMP_FORMATTER);
+        ZonedDateTime zonedDateTime;
+        try {
+            zonedDateTime = LocalDateTime.parse(timestamp, JSON_STATISTICS_TIMESTAMP_FORMATTER).atZone(UTC);
+        }
+        catch (DateTimeParseException _) {
+            // TODO: avoid this exception-driven logic
+            // When using `CREATE OR REPLACE TABLE` in Spark to change a column type from VARCHAR to TIMESTAMP,
+            // the checkpoint statistics possible stores the TIMESTAMP column in Spark's default format
+            // This fallback ensures compatibility with Spark's default TIMESTAMP format: 'yyyy-MM-dd HH:mm:ss',
+            // we read the precision to milliseconds by default.
+            return readPartitionTimestampWithZone(timestamp);
+        }
```

Comment:
I don't have a good grasp on the validity of this change.
Instead of this, I think we should modify `io.trino.plugin.deltalake.transactionlog.statistics.DeltaLakeJsonFileStatistics#deserializeStatisticsValue` to return `Optional#empty` when `deserializeColumnValue` throws an exception.
Doing that will resolve https://github.com/trinodb/trino/issues/24487 as well.

Reply:
I think we can just land https://github.com/trinodb/trino/pull/24489 and skip the first commit here

Reply:
yeah, that make sense

Reply:
you can rebase your PR to master now